snippets/simulate-onsets.r

```diff
@@ -0,0 +1,16 @@
+source(here::here("snippets", "load-ts.r"))
+source(here::here("functions", "censored-delay-pmf.r"))
+source(here::here("functions", "convolve-with-delay.r"))
+
+gen_time_pmf <- censored_delay_pmf(rgamma, max = 14, shape = 4, rate = 1)
+gen_time_pmf <- gen_time_pmf[-1] ## remove first element
+gen_time_pmf <- gen_time_pmf / sum(gen_time_pmf) ## renormalise
+
+ip_pmf <- censored_delay_pmf(rgamma, max = 14, shape = 5, rate = 1)
+onsets <- convolve_with_delay(inf_ts$infections, ip_pmf)
+onsets <- purrr::map_dbl(onsets, \(x) rpois(1, x))
```

Comment:
Not quite clear to me why we sample here instead of just using the delayed onsets later. I can see why we wouldn't model the expectation of the individual level process but earlier we motivated that as uncertainty from the count process - do we need to provide a different motivation here?

Reply:
Observation error? I think in general it makes sense to use a model that is a close the one we are using for fitting as possible

Reply:
also wow who knew it took a vector (well obviously you)

Reply:
> Observation error?

but we observe the delayed/truncated reports (and that's where we have a poisson in the fitting model) - what's the rationale for adding observation error twice (once at onset and then after delay)?

Reply:
@seabbs still not clear why we're sampling here btw if it's not the observation (as in the nowcasting session where we truncate and then add another observation error).